operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -45,52 +46,71 @@ func ValidateConfig(failOnMissingValidation bool, iopvalues map[string]interface
-		{"global.mtls.enabled", "the PeerAuthentication resource", nil},
-		{"global.mtls.auto", "meshConfig.enableAutoMtls", nil},
+		{"Values.global.certificates", "meshConfig.certificates", nil},
+		{"Values.global.trustDomainAliases", "meshConfig.trustDomainAliases", nil},
+		{"Values.global.outboundTrafficPolicy", "meshConfig.outboundTrafficPolicy", nil},
+		{"Values.global.localityLbSetting", "meshConfig.localityLbSetting", nil},
+		{"Values.global.policyCheckFailOpen", "meshConfig.policyCheckFailOpen", false},
+		{"Values.global.enableTracing", "meshConfig.enableTracing", false},
+		{"Values.global.proxy.accessLogFormat", "meshConfig.accessLogFormat", ""},
+		{"Values.global.proxy.accessLogFile", "meshConfig.accessLogFile", ""},
+		{"Values.global.proxy.accessLogEncoding", "meshConfig.accessLogEncoding", valuesv1alpha1.AccessLogEncoding_JSON},
+		{"Values.global.proxy.concurrency", "meshConfig.concurrency", uint32(0)},
+		{"Values.global.disablePolicyChecks", "meshConfig.disablePolicyChecks", nil},
+		{"Values.global.proxy.envoyAccessLogService", "meshConfig.envoyAccessLogService", nil},
+		{"Values.global.proxy.envoyMetricsService", "meshConfig.envoyMetricsService", nil},
+		{"Values.global.proxy.protocolDetectionTimeout", "meshConfig.protocolDetectionTimeout", ""},
+		{"Values.mixer.telemetry.reportBatchMaxEntries", "meshConfig.reportBatchMaxEntries", uint32(0)},
+		{"Values.mixer.telemetry.reportBatchMaxTime", "meshConfig.reportBatchMaxTime", ""},
+		{"Values.pilot.ingress", "meshConfig.ingressService, meshConfig.ingressControllerMode, and meshConfig.ingressClass", nil},
+		{"Values.global.mtls.enabled", "the PeerAuthentication resource", nil},
+		{"Values.global.mtls.auto", "meshConfig.enableAutoMtls", nil},
+		{"Values.grafana.enabled", "the samples/addons/ deployments", false},
+		{"Values.tracing.enabled", "the samples/addons/ deployments", false},
+		{"Values.kiali.enabled", "the samples/addons/ deployments", false},
+		{"Values.prometheus.enabled", "the samples/addons/ deployments", false},
+		{"AddonComponents.grafana.Enabled", "the samples/addons/ deployments", false},
+		{"AddonComponents.tracing.Enabled", "the samples/addons/ deployments", false},
+		{"AddonComponents.kiali.Enabled", "the samples/addons/ deployments", false},
+		{"AddonComponents.prometheus.Enabled", "the samples/addons/ deployments", false},
 	}
 	for _, d := range deprecations {
-		v, f, _ := tpath.GetFromStructPath(values, firstCharsToUpper(d.old))
+		// Grafana is a special case where its just an interface{}. A better fix would probably be defining
+		// the types, but since this is deprecated this is easier
+		v, f, _ := tpath.GetFromStructPath(iop, d.old)
 		if f && v != d.def {
-			messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s instead", d.old, d.new))
+			messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s instead", firstCharsToLower(d.old), d.new))
```

Comment:
we have a mix of "struct case" and "helm case", so instead we will let the inputs define the correct casing then just convert it to helm casing on output